### PR TITLE
cleanup genericapiserver authentication options

### PR DESCRIPTION
**Pull request by deads2k**
Moves client certificate CA to the authentication structs and adds front proxy for the delegation case.

@liggitt part of what you requested in the options pull.

### CODE REVIEW

File: pkg/genericapiserver/config.go [hunk 1]
```diff
@@ -307,6 +306,10 @@ func (c *Config) ApplyAuthenticationOptions(o *options.BuiltInAuthenticationOpti
 		return c
 	}
 
+	if o.ClientCert != nil && c.SecureServingInfo != nil {
+		c.SecureServingInfo.ClientCA = o.ClientCert.ClientCA
```
Comment (liggitt) [inline pkg/genericapiserver/config.go]: huh... we now have two potentially (hopefully?) distinct CA pools we'll validate client certs from (to validate request header proxy auth or to verify a client cert for auth directly). We need to include the full pool in the server handshake, so both files need to make it to the secure serving info

**Comment by liggitt** [comment]
comment on plumbing both pools to serving options, LGTM otherwise

File: examples/apiserver/apiserver.go [hunk 2]
```diff
@@ -99,14 +99,14 @@ func (serverOptions *ServerRunOptions) Run(stopCh <-chan struct{}) error {
...
-	config, err := genericapiserver.NewConfig().
+	config := genericapiserver.NewConfig().
 		ApplyOptions(serverOptions.GenericServerRunOptions).
-		ApplyInsecureServingOptions(serverOptions.InsecureServing).
-		ApplyAuthenticationOptions(serverOptions.Authentication).
-		ApplySecureServingOptions(serverOptions.SecureServing)
-	if err != nil {
+		ApplyInsecureServingOptions(serverOptions.InsecureServing)
+
+	if _, err := config.ApplySecureServingOptions(s.SecureServing); err != nil {
```
Comment (liggitt) [inline examples/apiserver/apiserver.go]: tests are complaining about this

**Comment by liggitt** [comment]
LGTM with the clientca bundle as a followup and with tests fixed

**Comment by deads2k** [comment]
@k8s-bot cvm gce e2e test this: issue #37940

**Comment by rmmh** [comment]
@k8s-bot gci gce e2e test this (job was accidentally deleted)

**Comment by rmmh** [comment]
@k8s-bot gci gce e2e test this (job was accidentally deleted)

**Comment by deads2k** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by deads2k** [comment]
@k8s-bot gci gce e2e test this

**Pull request merged**